improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -227,23 +227,91 @@ def test_exception_without_relative_to_threshold(self):
 
         msg = "Cube threshold coordinate does not define whether"
         with self.assertRaisesRegex(ValueError, msg):
-            self.plugin._ensure_monotonicity(self.forecast)
+            self.plugin._ensure_monotonicity_across_thresholds(self.forecast)
+
+
+class Test__combine_bin_pair_for_observation_frequency_monotonicity(unittest.TestCase):
+
+    """Test the _combine_bin_pair_for_observation_frequency_monotonicity method"""
+
+    def setUp(self):
+        """Set up forecast count"""
+        self.forecast_count = np.array([1000, 1000, 1000, 1000, 1000], dtype=np.float32)
+        self.forecast_probability_sum = np.array(
+            [0, 250, 500, 750, 1000], dtype=np.float32
+        )
+
+        self.expected_non_monotonic = np.array(
+            [
+                [0, 250, 500, 1750],  # Observation count
+                [0, 250, 500, 1750],  # Sum of forecast probability
+                [1000, 1000, 1000, 2000],  # Forecast count
+            ]
+        )
+
+    def test_monotonic(self):
+        """Test no bin pairs are combined, if all bin pairs are monotonic."""
+        obs_count = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        result = Plugin()._combine_bin_pair_for_observation_frequency_monotonicity(
+            obs_count, self.forecast_probability_sum, self.forecast_count
+        )
+        assert_array_equal(
+            result, [obs_count, self.forecast_probability_sum, self.forecast_count]
+        )
+
+    def test_one_non_monotonic_bin_pair(self):
+        """Test one bin pair is combined, if one bin pair is non-monotonic."""
+        obs_count = np.array([0, 250, 500, 1000, 750], dtype=np.float32)
+        result = Plugin()._combine_bin_pair_for_observation_frequency_monotonicity(
+            obs_count, self.forecast_probability_sum, self.forecast_count
+        )
+        assert_array_equal(result, self.expected_non_monotonic)
+
+    def test_two_non_monotonic_bin_pairs(self):
+        """Test one bin pair is combined, if two bin pairs are non-monotonic."""
+        obs_count = np.array([0, 750, 500, 1000, 750], dtype=np.float32)
+        self.expected_non_monotonic[0][1] = 750  # Amend observation count
+        result = Plugin()._combine_bin_pair_for_observation_frequency_monotonicity(
+            obs_count, self.forecast_probability_sum, self.forecast_count
+        )
+        assert_array_equal(result, self.expected_non_monotonic)
+
+
+class Test__assume_constant_observation_frequency(unittest.TestCase):
+    """Test the _assume_constant_observation_frequency method"""
+
+    def test_monotonic(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6001543)